pkg/nuctl/command/patch.go

```diff
@@ -0,0 +1,387 @@
+}
+
+func (c *patchCommandeer) initialize(ctx context.Context) error {
+
+	// parse the request timeout
+	if c.requestTimeout == "" {
+		c.requestTimeout = DefaultRequestTimeout
+	}
+	requestTimeoutDuration, err := time.ParseDuration(c.requestTimeout)
+	if err != nil {
+		return errors.Wrap(err, "Failed to parse request timeout")
+	}
+
+	// initialize http client
+	c.httpClient = &http.Client{
+		Timeout: requestTimeoutDuration,
+	}
+	if c.skipTLSVerify {
+		c.httpClient.Transport = &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		}
+	}
+
+	if err := c.initializePatchOptions(); err != nil {
+		return errors.Wrap(err, "Failed to initialize patch options")
+	}
+
+	c.rootCommandeer.loggerInstance.DebugWithCtx(ctx, "Initialized patch commandeer",
+		"patchOptions", c.patchOptions)
+
+	return nil
+}
+
+func (c *patchCommandeer) initializePatchOptions() error {
+	if len(c.patchOptionsMap) == 0 {
+		return nil
+	}
+
+	c.patchOptions = &resource.PatchOptions{}
+
+	// convert the patch options map to a patch options struct
+	if err := mapstructure.Decode(c.patchOptionsMap, c.patchOptions); err != nil {
+		return errors.Wrap(err, "Failed to decode patch options")
+	}
+
+	return nil
+}
+
+type patchFunctionsCommandeer struct {
+	*patchCommandeer
+
+	excludedProjects    []string
+	excludedFunctions   []string
+	concurrency         int
+	waitForFunction     bool
+	skipFunctionWithGPU bool
+	outputManifest      *nuctlcommon.OutputManifest
+}
+
+func newPatchFunctionsCommandeer(ctx context.Context, patchCommandeer *patchCommandeer) *patchFunctionsCommandeer {
+	commandeer := &patchFunctionsCommandeer{
+		patchCommandeer: patchCommandeer,
+		outputManifest:  nuctlcommon.NewOutputManifest(),
+	}
+
+	cmd := &cobra.Command{
+		Use:     "functions",
+		Aliases: []string{"func", "fn", "function"},
+		Short:   "patch functions",
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			// initialize root
+			if err := patchCommandeer.rootCommandeer.initialize(); err != nil {
+				return errors.Wrap(err, "Failed to initialize root")
+			}
+
+			// initialize patch commandeer
+			if err := patchCommandeer.initialize(ctx); err != nil {
+				return errors.Wrap(err, "Failed to initialize patch commandeer")
+			}
+
+			if err := commandeer.validateAndEnrichFlags(); err != nil {
+				return errors.Wrap(err, "Failed to validate flags")
+			}
+
+			return commandeer.patchFunctions(ctx)
+		},
+	}
+
+	cmd.PersistentFlags().StringSliceVarP(&commandeer.excludedProjects, "exclude-projects", "", []string{}, "Exclude projects to patch")
+	cmd.PersistentFlags().StringSliceVarP(&commandeer.excludedFunctions, "exclude-functions", "", []string{}, "Exclude functions to patch")
+	cmd.PersistentFlags().IntVarP(&commandeer.concurrency, "concurrency", "c", DefaultConcurrency, "Max number of parallel patches")
+	cmd.PersistentFlags().StringVarP(&commandeer.apiURL, "api-url", "", "", "URL of the nuclio API (e.g. https://nuclio.io:8070/api)")
```

Comment:
u alteady have that on root commandeer. why again? remove duplicate flags with root